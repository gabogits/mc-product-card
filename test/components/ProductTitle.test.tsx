import React from 'react';
import { ProductTitle } from '../../src/components/ProductTitle';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { product1 } from '../data/product';

describe('ProductTitle', () => {
  test('debe de mostrar el componente con el titulo personalizado', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Custom Product" className="custom-class" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostar el componente con el nombre', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => <ProductTitle className="custom-class" />}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
