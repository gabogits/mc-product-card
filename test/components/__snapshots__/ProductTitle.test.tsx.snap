// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente con el titulo personalizado 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;

exports[`ProductTitle debe mostar el componente con el nombre 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription custom-class"
  >
    Coffee - sin imagen
  </span>
</div>
`;
